core/src/main/java/org/infinispan/manager/DefaultCacheManager.java

```diff
@@ -687,18 +693,22 @@ private void terminate(String cacheName) {
    public void stop() {
       authzHelper.checkPermission(AuthorizationPermission.LIFECYCLE);
 
-      synchronized (this) {
-         if (stopping) {
-            log.trace("Ignore call to stop as the cache manager is stopping");
-            return;
+      //if start is in progress we need to wait until it's finished
+      startSemaphore.lock();
```

Comment:
well, but it's a lock for staring cache manager, not a cache - sorry, still unclear to me, how it helps